pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -309,6 +309,16 @@ type AlertmanagerSpec struct {
 	// by the given Alertmanager version.
 	// +optional
 	AdditionalArgs []Argument `json:"additionalArgs,omitempty"`
+
+	// Optional duration in seconds the pod needs to terminate gracefully.
+	// Value must be non-negative integer. The value zero indicates stop immediately via
+	// the kill signal (no opportunity to shut down) which may lead to data corruption.
+	//
+	// Defaults to 120 seconds.
```

Comment:
@simonpasquier  curious why we didn't add default in API validation but as a constant?

Reply:
In general, we don't use default in the API definition so I wanted to stay consistent. Default values are returned in the response of the Kubernetes API which may cause issues with some clients (e.g. they might interpret the value being present as a need for another update).

Reply:
We do have `// +kubebuilder:default:` set for many fields do you think we should remove it then for them?

Reply:
I don't think that we can remove any of them for compatibility reasons but I'd rather not add more :)